modules/osdk-publish-catalog.adoc

```diff
@@ -0,0 +1,168 @@
+// Module included in the following assemblies:
+//
+// * operators/operator_sdk/osdk-working-bundle-images.adoc
+
+[id="osdk-publish-catalog_{context}"]
+= Publishing a catalog containing a bundled Operator
+
+To install and manage Operators, Operator Lifecycle Manager (OLM) requires that Operator bundles are listed in an index image, which is referenced by a catalog on the cluster. As an Operator author, you can use the Operator SDK to create an index containing the bundle for your Operator and all of its dependencies. This is useful for testing on remote clusters and publishing to container registries.
+
+[NOTE]
+====
+The Operator SDK uses the `opm` CLI to facilitate index image creation. Experience with the `opm` command is not required. For advanced use cases, the `opm` command can be used directly instead of the Operator SDK.
+====
+
+.Prerequisites
+
+- Operator SDK CLI installed on a development workstation
+- Operator bundle image built and pushed to a registry
+- OLM installed on a Kubernetes-based cluster (v1.16.0 or later if you use `apiextensions.k8s.io/v1` CRDs, for example {product-title} {product-version})
+- Logged into the cluster with `oc` using an account with `cluster-admin` permissions
+
+.Procedure
+
+. Run the following `make` command in your Operator project directory to build an index image containing your Operator bundle:
++
+[source,terminal]
+----
+$ make catalog-build CATALOG_IMG=<registry>/<user>/<index_image_name>:<tag>
+----
++
+where the `CATALOG_IMG` argument references a repository that you have access to. You can obtain an account for storing containers at repository sites such as Quay.io.
+
+. Push the built index image to a repository:
++
+[source,terminal]
+----
+$ make catalog-push CATALOG_IMG=<registry>/<user>/<index_image_name>:<tag>
+----
++
+[TIP]
+====
+You can use Operator SDK `make` commands together if you would rather perform multiple actions in sequence at once. For example, if you had not yet built a bundle image for your Operator project, you can build and push both a bundle image and an index image with the following command:
+
+[source,terminal]
+----
+$ make bundle-build bundle-push catalog-build catalog-push \
+    BUNDLE_IMG=<bundle_image_pull_spec> \
+    CATALOG_IMG=<index_image_pull_spec>
+----
+====
+
+. Define a `CatalogSource` object that references the index image you just generated, and then create the object by using the `oc apply` command or web console:
++
+.Example `CatalogSource` YAML
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: cs-memcached
+  namespace: default
+spec:
+  displayName: My Test
+  publisher: Company
+  sourceType: grpc
+  image: quay.io/example/memcached-catalog:v0.0.1
```

Comment:
Should this image be "<index_image_pull_spec>"? Or notice that it should be the image <CATALOG_IMG>

Reply:
:+1: I've added a note about how this pull spec should match what was used earlier with `CATALOG_IMG`.